src/stan/lang/generator/generate_set_param_ranges.hpp

```diff
@@ -16,17 +17,27 @@ namespace stan {
      * determine the size required for the vector of param ranges and
      * the range for each parameter in the model by iterating over the
      * list of parameter variable declarations.
+     * Generated code is preceeded by stmt updating global variable
+     * `current_statement_begin__` to src file line number where
+     * parameter variable is declared.
      *
      * @param[in] var_decls sequence of variable declarations
+     * @param[in] indent indentation level
      * @param[in,out] o stream for generating
      */
     void generate_set_param_ranges(const std::vector<var_decl>& var_decls,
-                                   std::ostream& o) {
-      o << INDENT2 << "num_params_r__ = 0U;" << EOL;
-      o << INDENT2 << "param_ranges_i__.clear();" << EOL;
-      set_param_ranges_visgen vis(o);
-      for (size_t i = 0; i < var_decls.size(); ++i)
+                                   int indent, std::ostream& o) {
+      generate_indent(indent, o);
+      o << "num_params_r__ = 0U;" << EOL;
+      generate_indent(indent, o);
+      o << "param_ranges_i__.clear();" << EOL;
+      set_param_ranges_visgen vis(indent, o);
+      for (size_t i = 0; i < var_decls.size(); ++i) {
+        generate_indent(indent, o);
+        o << "current_statement_begin__ = " <<  var_decls[i].begin_line_ << ";"
```

Comment:
I think this generating line number thing should be pulled out into a function because you do it more than once.  It can be templated if the arguments all have `begin_line_` member variables.